From 60df6ccd27b22c8d28814e0d3f1dd30fa3cf69c5 Mon Sep 17 00:00:00 2001
From: Lad Prabhakar <prabhakar.mahadev-lad.rj@bp.renesas.com>
Date: Fri, 6 Sep 2024 16:57:43 +0100
Subject: [PATCH] clk: renesas: rcar-gen3: Fix the shifts calculated for MDSEL
 and PE clocks

Prior to patch 99c7e031196 ("clk: renesas: rcar-gen3: Replace SSCG caching
with MDSEL/PE caching") the shifts calculated for MDSEL and PE clocks are
as follows:

BIT(12) = 1 = shift 0
BIT(12) = 0 = shift 16

But after the patch 99c7e031196 ("clk: renesas: rcar-gen3: Replace SSCG
caching with MDSEL/PE caching") the shifts we caclulated as below:

BIT(12) = 1 = shift 16
BIT(12) = 0 = shift 0

This patch fixes the issue by restoring the proper shift values for MDSEL
and PE clock calculations.

Fixes: 99c7e031196 ("clk: renesas: rcar-gen3: Replace SSCG caching with MDSEL/PE caching")

Upstream-Status: Pending

Signed-off-by: Lad Prabhakar <prabhakar.mahadev-lad.rj@bp.renesas.com>
Signed-off-by: Chris Paterson <chris.paterson2@renesas.com>
---
 drivers/clk/renesas/clk-rcar-gen3.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/renesas/clk-rcar-gen3.c b/drivers/clk/renesas/clk-rcar-gen3.c
index b84024266f40..4f1dfbc174a8 100644
--- a/drivers/clk/renesas/clk-rcar-gen3.c
+++ b/drivers/clk/renesas/clk-rcar-gen3.c
@@ -69,7 +69,7 @@ static int gen3_clk_get_parent(struct gen3_clk_priv *priv, struct clk *clk,
 			return ret;
 
 		if (core->type == CLK_TYPE_GEN3_MDSEL) {
-			shift = priv->cpg_mode & BIT(core->offset) ? 16 : 0;
+			shift = priv->cpg_mode & BIT(core->offset) ? 0 : 16;
 			parent->dev = clk->dev;
 			parent->id = core->parent >> shift;
 			parent->id &= 0xffff;
@@ -318,7 +318,7 @@ static u64 gen3_clk_get_rate64(struct clk *clk)
 						"FIXED");
 
 	case CLK_TYPE_GEN3_MDSEL:
-		shift = priv->cpg_mode & BIT(core->offset) ? 16 : 0;
+		shift = priv->cpg_mode & BIT(core->offset) ? 0 : 16;
 		div = (core->div >> shift) & 0xffff;
 		rate = gen3_clk_get_rate64(&parent) / div;
 		debug("%s[%i] PE clk: parent=%i div=%u => rate=%llu\n",
-- 
2.34.1

